src/test/java/gov/cancer/tests/cts/advanced_search_form/CancerTypeSection_Test.java

```diff
@@ -0,0 +1,606 @@
+                                                                  String sideEffect, String expectedQueryParams ) {
+    TestRunner.run(AdvancedSearchPage.class, path, (AdvancedSearchPage page) -> {
+      page.getCancerTypeSection().scrollUntilAllButtonVisible();
+      //Click All button
+      page.getCancerTypeSection().getAllButton().click();
+      //selecting the Cancer Type
+      page.getCancerTypeSection().getCancerTypeCondition().selectItem(cancerType);
+      //selecting the Sub Type
+      page.getCancerTypeSection().getSubType().selectItem(subType);
+      //selecting the Stage
+      page.getCancerTypeSection().getStage().selectItem(stage);
+      //selecting the Side Effect
+      page.getCancerTypeSection().getSideEffects().selectItem(sideEffect);
+
+      // create a navigation event redirect to search results page
+      SearchNavigationResult search_result = page.clickSubmit(page.getFormAction().getFindTrialsButton());
+      //create a comparator object from the redirected URL query part
+      QueryParametersComparator querycomp = new QueryParametersComparator(search_result.getPageURL().getQuery());
+      //Assert that url contains expected query
+      Assert.assertTrue(querycomp.compareQueryParams(expectedQueryParams ),
+        " query parameters does not match for Cancer Type, SubType, Stage and Side effects");
+    });
+  }
+
+  /** Verify that when user inputs an invalid value to the Primary Cancer Type/Condition field,
+   * error message is displayed.
+   * parameters
+   * @param path       url
+   * @param cancerType cancer type value
+   * @param error error message
+   */
+  @Test(dataProvider = "getInvalidCancerType")
+  public void verifyInvalidCancerTypeEntries(String path, String cancerType, String error) {
+    TestRunner.run(AdvancedSearchPage.class, path, (AdvancedSearchPage page) -> {
+      page.getCancerTypeSection().scrollUntilAllButtonVisible();
+      //Click All button
+      page.getCancerTypeSection().getAllButton().click();
+      //selecting the Cancer Type
+      page.getCancerTypeSection().getCancerTypeCondition().enterText(cancerType);
+      //assert that message displayed is correct
+      Assert.assertEquals(page.getCancerTypeSection().getCancerTypeCondition().getSectionFieldMessage(),
+        error, "Correct message is not displayed for No available options");
+    });
+  }
+
+  /** Verify that when user inputs an invalid value to the SubType field,
+   * error message is displayed.
+   * parameters
+   * @param path       url
+   * @param cancerType cancer type value
+   * @param subType cancer sub type
+   * @param error error message
+   */
+  @Test(dataProvider = "getInvalidSubType")
+  public void verifyInvalidSubTypeEntries(String path, String cancerType,String subType, String error) {
+    TestRunner.run(AdvancedSearchPage.class, path, (AdvancedSearchPage page) -> {
```

Comment:
Same comment as above, and anywhere else the expected value doesn't have the word "expected" in the variable name.....

Reply:
fixed it